tensorflow/core/kernels/conv_ops_sycl.h

```diff
@@ -112,21 +136,40 @@ struct LaunchConv2DBackpropInputOp<SYCLDevice, T> {
                             stride_cols, out_rows,    out_cols,    pad_rows,
                             pad_cols};
 
+    // TODO: Remove the SYCLConv2DParams struct and always use the sycldnn one instead.
+    namespace sd = sycldnn::conv2d;
+    static_assert(sizeof(SYCLConv2DParams) == sizeof(sd::Conv2DParams),
+                  "SYCLConv2DParams shouldn't be used anymore");
+    auto sd_params = *reinterpret_cast<sd::Conv2DParams*>(&params);
+    sd::ConstantSelector<sd::Algorithm::Direct> selector;
+    auto device = context->eigen_device<SYCLDevice>();
+    sycldnn::backend::EigenBackend backend(device);
+
     T const* const in_ptr = out_backprop.template flat<T>().data();
     T const* const fil_ptr = filter.template flat<T>().data();
     T* const out_ptr = in_backprop->template flat<T>().data();
 
     SNN_SELECTOR sel;
```

Comment:
Could be in the branch?
```suggestion
    if(data_format == FORMAT_NCHW) {
      SNN_SELECTOR sel;
```